### ISSUE TITLE
OpenAPI 3.0 compatibility

### ISSUE THREAD

**Issue by joaobiriba** [Yelp/swagger-gradle-codegen#51]
It could be a good thing if the builder could parse OpenAPI 3.0 swagger json file

https://swagger.io/docs/specification/about/

**Comment by cortinico** [comment]
> It could be a good thing if the builder could parse OpenAPI 3.0 swagger json file

Hey @joaobiriba 
Thanks for the request. Ideally we should bump the Swagger Codegen version we are using to `3.x`.
Unfortunately this introduces a lot of breaking changes with the current [KotlinGenerator](https://github.com/Yelp/swagger-gradle-codegen/blob/master/plugin/src/main/java/com/yelp/codegen/KotlinGenerator.kt) class.

So ideally is something that will be implemented but not in the near future.
We're welcoming external contributors anyway :)

**Comment by joaobiriba** [comment]
Ok, let's see if I can found some time to work on it. This helped me to save some time, I could give back :)

**Comment by hybras** [comment]
So at this time the plugin does not support OpenAPI 3? Or is it json specifically?

**Comment by cortinico** [comment]
> So at this time the plugin does not support OpenAPI 3? Or is it json specifically?

Correct, at this time the plugin doesn't support OpenAPI 3. We might achieve this by bumping the `swagger-codegen` dependency to 3.x but we still have to investigate the amount of work needed.

**Comment by stephen-mojo** [comment]
It has been a while since the last comment on this thread. Has any work been done to bump Swagger Codegen to OpenAPI 3.0? Thanks.

**Comment by cortinico** [comment]
> Has any work been done to bump Swagger Codegen to OpenAPI 3.0? Thanks.

Not on my end. I'm unsure if @macisamuele did some work on that front.

**Comment by macisamuele** [comment]
I've looked into it long time ago (sorry for that) and seems that moving to the latest io.swagger:swagger-codegen (which supports openapi v3) might require some effort and it might interfere a bit (due to templating conflicts) with #103 .
I've sync a bit with @cortinico and we thought that I can start doing some prep work on enabling open api v3 and then afterwards dealing with eventual conflicts.

Enabling polymorphism seems being an interesting feature (at least it is for me) but we are not getting many requests on that regard, so it does not seem wise to hold openapi v3 longer

**Comment by jmgarbowski** [comment]
Hi everyone, is there any updates for this request?

**Comment by cortinico** [comment]
Sadly not. I haven't got much time to work on this recently. It's free to pick-up if someone wants to contribute. Me and @macisamuele were working on the Polymorphism support, and we would like to ship it before the OpenAPI change. Other than that, there are no major updates.

**Comment by CristianMG** [comment]
Hi, We have some idea, How much effort would take to give the support on OpenApi 3.0?
I would like to use this library, but in the back-end they use OpenApi 3.0

**Comment by aggarwalpulkit596** [comment]
@cortinico I would like to work on this could you maybe layout some ground tasks so that I can start?

**Comment by cortinico** [comment]
> So I suppose this lib is abandoned? :/

Just as a follow up: nope is not abandoned. We're doing on our free time that, as you can imagine, is extremely limited. Specifically I was working on this specific issue. The problem is that `swagger-codegen` 2.x => 3.x is introducing a significant number of API changes.

The result is that bumping this library and adding OpenAPI 3 support would require a significant amount of time.

If someone has time to look into this, and wants to send a PR, we'll be more than happy to review anyway 👍

**Comment by pischky** [comment]
Support for openAPI v3 would really be nice ;-)

**Comment by TheScriptan** [comment]
Bumping for OpenAPI v3 support:)

**Comment by alex-dokienko** [comment]
even in 2023 still valid thing to have 👍
